src/main/java/com/n1analytics/paillier/EncryptedNumber.java

```diff
@@ -416,4 +416,28 @@ public boolean equals(EncryptedNumber o) {
             context.equals(o.context) &&
             ciphertext.equals(o.ciphertext));
   }
+
+  /**
+   * Return a byte array representing the underlying raw number.
+   *
+   * WARNING: This method may return unencrypted information! Under no
+   * circumstances should the return value be sent anywhere.
+   *
+   * WARNING: This method is not intended for general use and may be
+   * removed in a future release. Think twice before using it.
+   */
+  public byte[] toByteArrayUnsafe() {
+    return ciphertext.toByteArray();
+  }
+
+  /**
+   * Return true or false according to whether this EncryptedNumber is
+   * safe for transmission.
+   *
+   * WARNING: This method is not intended for general use and may be
+   * removed in a future release. Think twice before using it.
+   */
+  public boolean getIsSafe() {
```

Comment:
At the moment I 

- take an array of EncryptedNumbers,
- unpack the ciphertexts,
- put them in a big byte array,
- send that to the GPU,
- have the GPU do its thing,
- retrieve the result from the GPU as a byte array,
- build the resulting ciphertexts, then
- _recreate the array of EncryptedNumbers_ using the new ciphertexts,
- return this new array as the result.

Regarding the last step, the EncryptedNumber constructor takes an _isSafe_ parameter which is _false_ by default (as it should be), so if I just recreate the EncryptedNumbers with the default, all the ones that were already safe will be needlessly (and expensively) re-encrypted. Since the calculation I just did on the GPU does not change the "safety" (in the case of the `inner` and `multiply` methods of DenseEncryptedVector), I obtain the original isSafe parameter using this `getIsSafe` method, and pass it to the constructor of EncryptedNumber when I construct the corresponding result.